presto-main/src/main/java/io/prestosql/eventlistener/EventListenerManager.java

```diff
@@ -55,59 +65,76 @@ public void addEventListenerFactory(EventListenerFactory eventListenerFactory)
         }
     }
 
-    public void loadConfiguredEventListener()
-            throws Exception
+    public void loadConfiguredEventListeners()
     {
-        File configFile = CONFIG_FILE.getAbsoluteFile();
-        if (!configFile.exists()) {
-            return;
+        checkState(loading.compareAndSet(false, true), "Event listeners already loaded");
+
+        List<File> configFiles = this.configFiles;
+        if (configFiles.isEmpty()) {
+            if (!CONFIG_FILE.exists()) {
+                return;
+            }
+            configFiles = ImmutableList.of(CONFIG_FILE);
         }
-
-        Map<String, String> properties = new HashMap<>(loadProperties(configFile));
-
-        String name = properties.remove(NAME_PROPERTY);
-        checkState(!isNullOrEmpty(name), "Access control configuration %s does not contain '%s'", configFile, NAME_PROPERTY);
-
-        setConfiguredEventListener(name, properties);
+        List<EventListener> eventListeners = configFiles.stream()
+                .map(this::createEventListener)
+                .collect(toImmutableList());
+        this.configuredEventListeners.set(eventListeners);
     }
 
-    @VisibleForTesting
-    protected void setConfiguredEventListener(String name, Map<String, String> properties)
+    private EventListener createEventListener(File configFile)
     {
-        requireNonNull(name, "name is null");
-        requireNonNull(properties, "properties is null");
-
-        log.info("-- Loading event listener --");
-
-        EventListenerFactory eventListenerFactory = eventListenerFactories.get(name);
-        checkState(eventListenerFactory != null, "Event listener '%s' is not registered", name);
-
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(eventListenerFactory.getClass().getClassLoader())) {
-            EventListener eventListener = eventListenerFactory.create(ImmutableMap.copyOf(properties));
-            this.configuredEventListener.set(Optional.of(eventListener));
+        log.info("-- Loading event listener %s --", configFile);
+        configFile = configFile.getAbsoluteFile();
+        Map<String, String> properties;
```

Comment:
This could be extracted as method so you don't need to have uninitialized variable.